Wrappers/Python/cil/optimisation/operators/BlockOperator.py

```diff
@@ -63,139 +66,143 @@ def __init__(self, *args, **kwargs):
+        '''
+        return [op.norm() for op in self.operators]
+
+    def set_norms(self, norms):
+        '''Uses the set_norm() function in Operator to set the norms of the operators in the BlockOperator from a list of custom values. 
+        '''
+        if len(norms) != len(self):
+            raise ValueError(
+                "The length of the list of norms should be equal to the number of operators in the BlockOperator")
+
+        for j, value in enumerate(norms):
+            self.operators[j].set_norm(value)
+
     def direct(self, x, out=None):
         '''Direct operation for the BlockOperator
 
         BlockOperator work on BlockDataContainer, but they will work on DataContainers
         and inherited classes by simple wrapping the input in a BlockDataContainer of shape (1,1)
         '''
-        
-        if not isinstance (x, BlockDataContainer):
+
+        if not isinstance(x, BlockDataContainer):
             x_b = BlockDataContainer(x)
         else:
             x_b = x
         shape = self.get_output_shape(x_b.shape)
         res = []
-        
+
         if out is None:
-        
+
             for row in range(self.shape[0]):
                 for col in range(self.shape[1]):
                     if col == 0:
-                        prod = self.get_item(row,col).direct(x_b.get_item(col))
+                        prod = self.get_item(row, col).direct(
+                            x_b.get_item(col))
                     else:
-                        prod += self.get_item(row,col).direct(x_b.get_item(col))
+                        prod += self.get_item(row,
+                                              col).direct(x_b.get_item(col))
                 res.append(prod)
             return BlockDataContainer(*res, shape=shape)
-                
+
         else:
-            
+
             tmp = self.range_geometry().allocate()
             for row in range(self.shape[0]):
                 for col in range(self.shape[1]):
-                    if col == 0:       
-                        self.get_item(row,col).direct(
-                                                      x_b.get_item(col),
-                                                      out=out.get_item(row))                        
+                    if col == 0:
+                        self.get_item(row, col).direct(
+                            x_b.get_item(col),
+                            out=out.get_item(row))
                     else:
                         a = out.get_item(row)
-                        self.get_item(row,col).direct(
-                                                      x_b.get_item(col), 
-                                                      out=tmp.get_item(row))
+                        self.get_item(row, col).direct(
+                            x_b.get_item(col),
+                            out=tmp.get_item(row))
                         a += tmp.get_item(row)
```

Comment:
What is `a` used for?

Reply:
Definitely unclear but seems to work. Have added issue #1544